addon/-private/system/store.js

```diff
@@ -2429,9 +2429,10 @@ Store = Service.extend({
     // payload push.  In the common case where we are pushing many more
     // instances than types we want to minimize the cost of looking up the
     // inverse map and the overhead of Ember.get adds up.
-    let modelNameToInverseMap = Object.create(null);
+    let modelNameToInverseMap;
 
     for (let i = 0, l = pushed.length; i < l; i += 2) {
+      modelNameToInverseMap = modelNameToInverseMap || Object.create(null);
```

Comment:
Is it worth preventing the `ToBoolean` conversion here e.g. `(modelNameToInverseMap !== undefined)`

Reply:
I believe the cost of `ToBoolean` coercion in this context is negligible. I could default `modelNameToInverseMap` to `false` though, if you would prefer. That would be a minimal enough change.  Let me know

Reply:
No this seems fine.